### PR TITLE
Circuit Labels

### PR DESCRIPTION
**Pull request by r-esistor**
Everything looks to be working, but this is also my first PR so there's a solid chance I fucked up somewhere

:cl: Resistor
add: Added circuit labeling! Hit your assemblies with a pen to write down instructions or warnings.
/:cl:

[why]: 

More complex circuits aren't always user-friendly, this makes it easier to hand circuits out to crew without having to also give them a lecture on proper usage.

### CODE REVIEW

**Comment by arsserpentarium** [comment]
Label can't be saved in blueprints this way.
Still good change.

**Comment by r-esistor** [comment]
I'll look into making it appear in the blueprint text, not 100% sure how it works right now

**Comment by AverageJoe82** [comment]
circuit information is stored as json if you know what that is

File: code/modules/integrated_electronics/core/assemblies.dm [hunk 1]
```diff
@@ -10,6 +10,7 @@
 	icon_state = "setup_small"
 	flags_1 = NOBLUDGEON_1
 	materials = list()		// To be filled later
+	var/label = ""
```
Comment (Cyberboss) [inline code/modules/integrated_electronics/core/assemblies.dm]: just do var/label so it stays as null

File: code/modules/integrated_electronics/core/assemblies.dm [hunk 2]
```diff
@@ -56,6 +57,10 @@
 
 /obj/item/electronic_assembly/examine(mob/user)
 	. = ..()
+	if(label != "")
```
Comment (Cyberboss) [inline code/modules/integrated_electronics/core/assemblies.dm]: if(!label)

File: code/modules/integrated_electronics/core/assemblies.dm [hunk 3]
```diff
@@ -56,6 +57,10 @@
...
+		to_chat(user, "The label reads: " + label)
```
Comment (Cyberboss) [inline code/modules/integrated_electronics/core/assemblies.dm]: Prefer "The label reads: [label]"

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
Why is this not using unique_rename or whatever it's been converted into now?

**Comment by r-esistor** [comment]
Currently, circuit renaming is built into the assembly interface, so changing it to unique_rename would I think prevent names from being carried through analyzer blueprints (not 100% sure though)

**Comment by optimumtact** [comment]
Is everything alright, do you need some assistance?

Also quick note, in general, it's a good idea to not use the master branch for a change, you should make a new feature branch to develop in, based off the master.

`git checkout -b mynewfeature master`

That way if for some reason a PR isn't accepted you're not left with commits stuck on your branch that you then have to remove for future changes

**Comment by optimumtact** [comment]
Travis is failing your PR due to a compile error in your code (one advantage of developing locally is you can compile locally and catch these yourself)

```
tgstation.mdme:2619:error: unterminated text expression (expecting ])
tgstation.mdme:2619:error: end of file: expected )
tgstation.mdme:2619:error: end of file: expected }
code/modules/integrated_electronics/core/assemblies.dm:61:error: location of top-most unmatched {
tgstation.mdme:2619:error: end of file: expected }
code/modules/integrated_electronics/core/assemblies.dm:59:error: location of top-most unmatched {
tgstation.mdme.dmb - 6 errors, 0 warnings (4/30/18 10:39 pm)
```

File: code/modules/integrated_electronics/core/assemblies.dm [hunk 4]
```diff
@@ -56,6 +57,10 @@
 
 /obj/item/electronic_assembly/examine(mob/user)
 	. = ..()
+	if(!label)
+		to_chat(user, "The label reads: [label")
```
Comment (optimumtact) [inline code/modules/integrated_electronics/core/assemblies.dm]: missing ] here will be the cause

**Comment by optimumtact** [comment]
oh and uh, finally, welcome to the repository :smile: if you get stuck with anything you can always reach us in the irc (#coderbus on irc.rizon.net) or if that's not your speed, there is a coding channel in the discord https://discord.gg/SumR2HU (although if I'm honest we check the IRC more often for people)

**Comment by ExcessiveUseOfCobblestone** [comment]
@r-esistor that's correct but it's an already built feature and with the time you would have to spend rebuilding your circuits you'd still come out on top if you had to ctrl+c ctrl+v the name.

**Comment by r-esistor** [comment]
@optimumtact Probably should've noticed the missing bracket ☹️ Bit new to git, I'll start using a local editor/proper branches/etc once I get back to my desktop

@ExcessiveUseOfCobblestone Sounds good! I'll rework it to use unique_rename, hopefully someone with a bit more experience in IC code can make it work with analyzer blueprints

**Comment by r-esistor** [comment]
I'm just going to start again with proper technique this time

**Comment by optimumtact** [comment]
don't give up, I believe in you

**Pull request closed without merging**